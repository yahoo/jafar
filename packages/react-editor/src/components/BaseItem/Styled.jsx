/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import styled from 'styled-components';

const Wrapper = styled.div`
  flex: 1;
  height: 100%;
  display: flex;
  flex-direction: row;

  > div {
    flex: 1;
  }
`;

const JsonViewWrapper = styled.div`
  overflow-y: auto;
  overflow-x: auto;
  max-width: 500px;
  border-left: 1px solid #e1e1e1;
  background-color: rgb(252, 253, 253);
`;

export default {
  Wrapper,
  JsonViewWrapper,
};
