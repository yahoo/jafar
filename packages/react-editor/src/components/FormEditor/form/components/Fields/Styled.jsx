/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import styled from 'styled-components';
import { withTheme } from '@material-ui/styles';

export const BooleanWrapper = styled.div`
  position: relative;
  top: 7px;
`;

export const FieldLink = withTheme(styled.a`
  flex: 1;
  text-decoration: none;
  cursor: pointer;
  color: ${props => props.theme.palette.secondary.main};
  &:hover {
    text-decoration: underline;
  }
`);

export const FieldEditorWrapper = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 21;
`;
