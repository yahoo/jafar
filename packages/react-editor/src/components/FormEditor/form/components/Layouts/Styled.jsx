/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import styled from 'styled-components';

export const Wrapper = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 21;
`;

export const GridWrapper = styled.div`
  max-height: 400px;
  display: flex;
  flex-direction: column;
  flex: 1;
`;
