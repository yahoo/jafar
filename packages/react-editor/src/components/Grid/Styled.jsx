/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import styled from 'styled-components';
import Button from '@material-ui/core/Button';

export const HeaderButton = styled(Button)`
  margin-left: 10px !important;
`;

export const Grid = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  overflow: hidden;
`;

export const Row = styled.div`
  display: flex;
  flex-direction: row;
  line-height: 40px;
  [aria-label="actions"] {
    visibility: hidden;
  }
  &:hover {
    [aria-label="actions"] {
      visibility: visible;
    }
  }
  &:nth-child(even) {
    background-color: #f9f9f9;
  }
  &:nth-child(odd) {
    background-color: #f1f1f1;
  }
`;

export const HeaderRow = styled(Row)`
  font-weight: bold;
  background-color: #e0e0e0 !important;
`;

export const Cell = styled.div`
  padding: 10px 20px;
  flex: ${props => props.minWidth ? 'unset' : 1 };
  min-width: ${props => props.minWidth};
  overflow: hidden;
`;

export const Body = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  overflow-y: auto;
`;

export const HeaderMenu = styled.div`
  text-align: right;
  padding-bottom: 10px;
`;
