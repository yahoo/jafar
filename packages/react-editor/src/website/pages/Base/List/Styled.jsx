import styled from 'styled-components';
import { withTheme } from '@material-ui/styles';

export const Wrapper = styled.div`
  padding: 40px;
  flex: 1;
  display: flex;
  flex-direction: column;
  overflow: hidden;
`;

export const Title = styled.h1`
  font-weight: 500;
`;

export const BooleanWrapper = styled.div`
  position: relative;
  top: 7px;
`;

export const Link = withTheme(styled.a`
  flex: 1;
  text-decoration: none;
  cursor: pointer;
  color: ${props => props.theme.palette.secondary.main};
  &:hover {
    text-decoration: underline;
  }
`);
