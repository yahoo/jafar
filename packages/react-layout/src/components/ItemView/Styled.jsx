/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import styled from 'styled-components';

export default {
  Wrapper: styled.div`
    color: #3d3d3d;
    background-color: #fff;
    height:100%;
    display: flex;
    flex-direction: column;
  `,
  Header: styled.div`
    flex: 0 0 auto;
    padding: 0 50px;
    border-bottom: 3px solid #e1e1e1;
    position: relative;
    min-height: 48px;
    position: relative;
  `,
  OptionsWrapper: styled.div`
    position: absolute;
    right: 15px;
    bottom: 0;
  `,
  Title: styled.h3`
    font-size: 28px;
    font-weight: 500;
    margin: 0;
    padding: ${props => (props.hasTabs ? '25px 0 10px 0' : '25px 0')};
  `,
  TabsWrapper: styled.div`
    position: relative;
    top: 3px;
    left: -10px;
  `,
  Sections: styled.div`
    flex-grow: 1;
    overflow-y: auto;
  `,
};
