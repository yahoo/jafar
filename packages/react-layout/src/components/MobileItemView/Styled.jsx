/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import styled from 'styled-components';

const mobileItemViewSizeCache = {};

export default (size) => {
  const key = size;
  if (!mobileItemViewSizeCache[key]) {
    mobileItemViewSizeCache[key] = {
      Wrapper: styled.div`
        color: #3d3d3d;
        background-color: #fff;
        height:100%;
        display: flex;
        flex-direction: column;
      `,
      Header: styled.div`
        background-color: #f0f0f0;
        font-size: 16px;
        padding: 10px 10px 10px ${10 + (size * 10)}px;
        display: flex;
        flex-direction: row;
        flex: 0 0 auto;
      `,
      Title: styled.div`
        font-size: 18px;
        font-weight: bold;
        flex: 1;
        line-height: 45px;
      `,
      Sections: styled.div`
        flex: 1;
        overflow-y: auto;
      `,
    };
  }
  return mobileItemViewSizeCache[key];
};

