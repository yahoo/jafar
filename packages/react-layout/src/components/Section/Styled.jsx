/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import styled from 'styled-components';

const sectionSizeCache = {};

export default (size, root) => {
  size = Math.max(size, 0);
  const key = `${size},${root}`;

  if (!sectionSizeCache[key]) {
    sectionSizeCache[key] = {
      Wrapper: styled.div`
        padding: ${() => (root ? `${(size * 8) + 18}px ${(size * 10) + 10}px ${size * 10}px ${(size * 10) + 10}px` : '0')};
        border-bottom: ${() => (root ? '1px solid #d8d8d8' : 'none')};
        &:last-child {
          border-bottom: none;
        }
      `,
      Title: styled.div`
        text-transform: uppercase;
        line-height: 1.1;
        color: inherit;
        font-size: ${() => (14 + (2 * size))}px;
        font-weight: ${() => root ? '500' : '400' };
        padding-bottom: ${() => (Math.max((size * 10) + 10, 24))}px;
        margin: 0;
      `,
    };
  }
  return sectionSizeCache[key];
};

