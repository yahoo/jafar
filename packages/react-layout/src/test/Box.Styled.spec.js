/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import React from 'react';
import renderer from 'react-test-renderer';
import Styled from '../components/Box/Styled';
import 'jest-styled-components';

describe('Styled', () => {
  describe('Row', () => {
    it('Should render ok', () => {
      const tree = renderer.create(
        <Styled.Row />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Column', () => {
    it('Should render ok', () => {
      const tree = renderer.create(
        <Styled.Column />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('Should render ok - with style', () => {
      const tree = renderer.create(
        <Styled.Column style={{ width: '400px', maxWidth: '400px', margin: '50px 0' }} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
