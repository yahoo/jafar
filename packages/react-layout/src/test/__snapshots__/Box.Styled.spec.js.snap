// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Column Should render ok - with style 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
  style={
    Object {
      "margin": "50px 0",
      "maxWidth": "400px",
      "width": "400px",
    }
  }
/>
`;

exports[`Styled Column Should render ok 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`Styled Row Should render ok 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
  className="c0"
/>
`;
